File: frontend/src/api/people.ts
import apiClient from './index'

// ─── Types ────────────────────────────────────────────────────────────────────

export interface CommunityRoleOut {
  id: number
  community_name: string
  project_url: string | null
  role: string
  role_label: string | null
  is_current: boolean
  started_at: string | null
  ended_at: string | null
  source_url: string | null
}

export interface PersonListOut {
  id: number
  display_name: string
  avatar_url: string | null
  github_handle: string | null
  email: string | null
  company: string | null
  source: string
  created_at: string
  community_names: string[]
}

export interface PersonOut {
  id: number
  display_name: string
  avatar_url: string | null
  github_handle: string | null
  gitcode_handle: string | null
  email: string | null
  phone: string | null
  company: string | null
  location: string | null
  bio: string | null
  tags: string[]
  notes: string | null
  source: string
  created_by_id: number | null
  created_at: string
  updated_at: string
  community_roles: CommunityRoleOut[]
}

export interface PersonCreate {
  display_name: string
  avatar_url?: string | null
  github_handle?: string | null
  gitcode_handle?: string | null
  email?: string | null
  phone?: string | null
  company?: string | null
  location?: string | null
  bio?: string | null
  tags?: string[]
  notes?: string | null
  source?: string
}

export interface PersonUpdate {
  display_name?: string
  avatar_url?: string | null
  github_handle?: string | null
  gitcode_handle?: string | null
  email?: string | null
  phone?: string | null
  company?: string | null
  location?: string | null
  bio?: string | null
  tags?: string[]
  notes?: string | null
}

export interface CommunityRoleCreate {
  community_name: string
  project_url?: string | null
  role: string
  role_label?: string | null
  is_current?: boolean
  started_at?: string | null
  ended_at?: string | null
  source_url?: string | null
}

export interface PaginatedPeople {
  items: PersonListOut[]
  total: number
  page: number
  page_size: number
}

// ─── People CRUD ──────────────────────────────────────────────────────────────

export async function listPeople(params?: {
  q?: string
  tag?: string
  company?: string
  source?: string
  page?: number
  page_size?: number
}) {
  const res = await apiClient.get<PaginatedPeople>('/people', { params })
  return res.data
}

export async function createPerson(data: PersonCreate) {
  const res = await apiClient.post<PersonOut>('/people', data)
  return res.data
}

export async function getPerson(id: number) {
  const res = await apiClient.get<PersonOut>(`/people/${id}`)
  return res.data
}

export async function updatePerson(id: number, data: PersonUpdate) {
  const res = await apiClient.patch<PersonOut>(`/people/${id}`, data)
  return res.data
}

export async function deletePerson(id: number) {
  await apiClient.delete(`/people/${id}`)
}

// ─── Community Roles ─────────────────────────────────────────────────────────

export async function listPersonRoles(personId: number) {
  const res = await apiClient.get<CommunityRoleOut[]>(`/people/${personId}/roles`)
  return res.data
}

export async function addPersonRole(personId: number, data: CommunityRoleCreate) {
  const res = await apiClient.post<CommunityRoleOut>(`/people/${personId}/roles`, data)
  return res.data
}

File: backend/app/schemas/people.py
from datetime import date, datetime

from pydantic import BaseModel


class CommunityRoleCreate(BaseModel):
    community_name: str
    project_url: str | None = None
    role: str
    role_label: str | None = None
    is_current: bool = True
    started_at: date | None = None
    ended_at: date | None = None
    source_url: str | None = None


class CommunityRoleOut(BaseModel):
    id: int
    community_name: str
    project_url: str | None
    role: str
    role_label: str | None
    is_current: bool
    started_at: date | None
    ended_at: date | None
    source_url: str | None

    model_config = {"from_attributes": True}


class PersonCreate(BaseModel):
    display_name: str
    avatar_url: str | None = None
    github_handle: str | None = None
    gitcode_handle: str | None = None
    email: str | None = None
    phone: str | None = None
    company: str | None = None
    location: str | None = None
    bio: str | None = None
    tags: list[str] = []
    notes: str | None = None
    source: str = "manual"


class PersonUpdate(BaseModel):
    display_name: str | None = None
    avatar_url: str | None = None
    github_handle: str | None = None
    gitcode_handle: str | None = None
    email: str | None = None
    phone: str | None = None
    company: str | None = None
    location: str | None = None
    bio: str | None = None
    tags: list[str] | None = None
    notes: str | None = None


class PersonOut(BaseModel):
    id: int
    display_name: str
    avatar_url: str | None
    github_handle: str | None
    gitcode_handle: str | None
    email: str | None
    phone: str | None
    company: str | None
    location: str | None
    bio: str | None
    tags: list[str]
    notes: str | None
    source: str
    created_by_id: int | None
    created_at: datetime
    updated_at: datetime
    community_roles: list[CommunityRoleOut] = []

    model_config = {"from_attributes": True}


class PersonListOut(BaseModel):
    id: int
    display_name: str
    avatar_url: str | None
    github_handle: str | None
    email: str | None
    company: str | None
    source: str
    created_at: datetime
    community_names: list[str] = []

    model_config = {"from_attributes": True}


class PaginatedPeople(BaseModel):
    items: list[PersonListOut]
    total: int
    page: int
    page_size: int


class MergeConfirm(BaseModel):
    """确认/拒绝去重匹配"""
    import_row: dict
    person_id: int | None  # None 表示创建新档案
    confirmed: bool
